src/hooks/api.ts

```diff
@@ -0,0 +1,72 @@
+type Method = 'GET' | 'POST' | 'PUT' | 'DELETE';
+
+interface ApiOptions {
+  method?: Method;
+  body?: Record<string, unknown>;
+  headers?: Record<string, string>;
+  token?: string;
+}
+
+interface ApiError extends Error {
+  flash?: Record<string, unknown>;
+  statusCode?: number;
+}
+
+const API_BASE_URL = process.env.NEXT_PUBLIC_BACKEND_URL;
+
+export const apiRequest = async <T>(
+  url: string,
+  { method = 'GET', body, headers = {}, token }: ApiOptions = {}
+): Promise<T> => {
+  const defaultHeaders: Record<string, string> = {
+    'Content-Type': 'application/json',
+  };
+
+  const authHeaders: Record<string, string> = token
+    ? { Authorization: `Bearer ${token}` }
+    : {};
```

Comment:
Retrieve the token (if it exists) from the browser cookies and set it to `options.headers.authorization`. Create a `getCookie` utility function to retrieve the cookie holding the token by its key, `connect.sid`, from the browser.

```suggestion
   options.headers.authorization = getCookie('connect.sid');
```